### ISSUE TITLE
[Good First Issue]: Introduce testing on Android in GitHub Actions

### ISSUE THREAD

**Issue by akashchi** [openvinotoolkit/openvino#26991]
### Context

At the moment, we have two GitHub Actions workflows in which we cross-compile OpenVINO artefacts using Ubuntu 22 for Android [x64](https://github.com/openvinotoolkit/openvino/blob/3f70084d336039e93715443699610d94065d5389/.github/workflows/android_x64.yml) and [arm64](https://github.com/openvinotoolkit/openvino/blob/3f70084d336039e93715443699610d94065d5389/.github/workflows/android_arm64.yml). 

We would like to enable testing of the produced artefacts on an Android emulator in GitHub Actions.

### What needs to be done?

The idea is to implement test jobs similarly to [existing test jobs in other workflows](https://github.com/openvinotoolkit/openvino/blob/master/docs/dev/ci/github_actions/adding_tests.md#creating-a-new-job):
1. Enable building of tests via `ENABLE_TESTS=ON` in the cmake configuration step: https://github.com/openvinotoolkit/openvino/blob/3f70084d336039e93715443699610d94065d5389/.github/workflows/android_x64.yml#L124
2. [Pack](https://github.com/openvinotoolkit/openvino/blob/3f70084d336039e93715443699610d94065d5389/.github/workflows/job_build_linux.yml#L182) and [upload](https://github.com/openvinotoolkit/openvino/blob/3f70084d336039e93715443699610d94065d5389/.github/workflows/job_build_linux.yml#L253) build and test artefacts
3. [Add a test job](https://github.com/openvinotoolkit/openvino/blob/master/docs/dev/ci/github_actions/adding_tests.md#creating-a-new-job) that utilises an Android emulator with the build and test artefacts from the previous step
4. Run any of the produced test binaries within the Android emulator using the [ADB tool](https://github.com/openvinotoolkit/openvino/blob/master/docs/dev/build_android.md#use-the-adb-tool-to-run-the-example-model-and-check-the-library)

Many Android emulators are available:
* via Docker images:
  * https://hub.docker.com/r/cimg/android/tags
  * https://github.com/budtmo/docker-android
* via GitHub Actions
  * https://github.com/ReactiveCircus/android-emulator-runner

> Note: the list is not extensive and if you find any other ways to run an Android emulator in GitHub Actions, please do not hesitate to use it.

In OpenVINO, we use [pre-built Docker images](https://github.com/openvinotoolkit/openvino/blob/master/docs/dev/ci/github_actions/docker_images.md#custom-docker-images-and-handle_docker-action) for building and testing so it is recommended that you use a Docker image with an Android emulator for testing.

Refer to [the OpenVINO GitHub Actions documentation](https://github.com/openvinotoolkit/openvino/tree/3f70084d336039e93715443699610d94065d5389/docs/dev/ci/github_actions) for more details on the GitHub Actions setup.

### Resources

- [Contribution guide - start here!](https://github.com/openvinotoolkit/openvino/blob/master/CONTRIBUTING.md)
- [Intel DevHub Discord channel](https://discord.gg/7pVRxUwdWG) - engage in discussions, ask questions and talk to OpenVINO developers
- [How to link your Pull Request to an issue](https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue#manually-linking-a-pull-request-to-an-issue-using-the-pull-request-sidebar)
- [OpenVINO GitHub Actions documentation](https://github.com/openvinotoolkit/openvino/tree/3f70084d336039e93715443699610d94065d5389/docs/dev/ci/github_actions)

### Contact points

If you have any questions, please contact @akashchi and @mryzhov. 

If you create a Pull Request, please add @akashchi and @mryzhov as reviewers and assignees.

### Ticket

149906

**Comment by geeky33** [comment]
.take

**Comment by geeky33** [comment]
hi @darksapien23151 due to some health issues going on from 1-2 days i have just took a small break.. i will be back from tomorrow 

there are some errors occurring hence fixing those at the same time.
thanks.

**Comment by dorloff** [comment]
@geeky33 , are you still planning to work on it?

**Comment by geeky33** [comment]
@dorloff yes sir i m discussing the issue with the mentor

**Comment by p-wysocki** [comment]
I'm unassigning the task due to assignee's inactivity.

**Comment by p-wysocki** [comment]
Hello @darksapien23151, are you still working on this?
bump @akashchi

**Comment by darksapien23151** [comment]
Yes, @p-wysocki , just that I have to get a clarification on this!


> Hey! [@akashchi](https://github.com/akashchi) I’m working on an Android emulator test job and realized that GitHub runners support Android emulators only on macOS, while we use Ubuntu for cross-compilation. I’d like to learn more about best practices for handling this. Should we split jobs and transfer artifacts, or is there a better approach? Any insights would be super helpful!

**Comment by akashchi** [comment]
Hi @darksapien23151, at the moment we do not want to use the GitHub-hosted runners for anything other than light jobs (like lint, etc.). We have the self-hosted Windows and Linux runners available in the GitHub Actions and are preferred for the build and test jobs.
You can find more information in the documentation: https://github.com/openvinotoolkit/openvino/tree/f4d0fd77c717bd055aab4a766d3f2bc9820db0be/docs/dev/ci/github_actions. Additional software (like the Android emulator) could be installed and configured in the runtime, i.e., in the job or in the Dockerfile for Linux runners.

**Comment by darksapien23151** [comment]
I understand, thank you for sharing it! will look into it and get back soon.

**Comment by devesh-047** [comment]
.take
